ciongke/github/client.go

```diff
@@ -0,0 +1,226 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package github
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+)
+
+type Client struct {
+	client *http.Client
+	token  string
+	base   string
+	dry    bool
+}
+
+const githubBase = "https://api.github.com"
+
+// NewClient creates a new fully operational GitHub client.
+func NewClient(token string) *Client {
+	return &Client{
+		client: &http.Client{},
+		token:  token,
+		base:   githubBase,
+		dry:    false,
+	}
+}
+
+// NewDryRunClient creates a new client that will not perform mutating actions
+// such as setting statuses or commenting, but it will still query GitHub and
+// use up API tokens.
+func NewDryRunClient(token string) *Client {
+	return &Client{
+		client: &http.Client{},
+		token:  token,
+		base:   githubBase,
+		dry:    true,
+	}
+}
+
+func (c *Client) request(method, path string, body io.Reader) (*http.Request, error) {
```

Comment:
Also, consider making the body an `interface{}` and moving the json.Marshal inside the request object, since all bodies are JSON-encoded.